File: app/src/main/java/pl/expert/mobilewzr/data/converter/CsvConverterFactory.java
package pl.expert.mobilewzr.data.converter;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Converter;
import retrofit2.Retrofit;

import java.lang.annotation.Annotation;
import java.lang.reflect.Type;

public final class CsvConverterFactory extends Converter.Factory {

    public static CsvConverterFactory create() {
        return new CsvConverterFactory();
    }

    private CsvConverterFactory() {
    }

    @Override
    public Converter<ResponseBody, ?> responseBodyConverter(Type type, Annotation[] annotations, Retrofit retrofit) {
        return new CsvResponseBodyConverter();
    }

    @Override
    public Converter<?, RequestBody> requestBodyConverter(Type type, Annotation[] parameterAnnotations, Annotation[] methodAnnotations, Retrofit retrofit) {
        return super.requestBodyConverter(type, parameterAnnotations, methodAnnotations, retrofit);
    }
}

File: app/src/main/java/pl/expert/mobilewzr/data/converter/CsvConverterFactory.kt
package pl.expert.mobilewzr.data.converter

import okhttp3.ResponseBody
import retrofit2.Converter
import retrofit2.Retrofit
import java.lang.reflect.Type

class CsvConverterFactory : Converter.Factory() {

    override fun responseBodyConverter(
        type: Type, annotations: Array<Annotation>, retrofit: Retrofit
    ): Converter<ResponseBody, *>? {
        return CsvResponseBodyConverter()
    }

}